pkg/backend/path_creds.go

```diff
@@ -81,15 +81,15 @@ func (b *backend) credsUpdateOperation(ctx context.Context, req *logical.Request
 		}
 
 		tok, err = cb.Build().Exchange(ctx, code.(string))
-		if _, ok := err.(*oauth2.RetrieveError); ok {
-			return logical.ErrorResponse("invalid code"), nil
+		if rErr, ok := err.(*oauth2.RetrieveError); ok {
+			return logical.ErrorResponse("invalid code: %v", (*oauth2.RetrieveError)(rErr)), nil
```

Comment:
Note that your error here is already of type `*oauth2.RetrieveError` so you don't need to cast it again.